app/src/main/java/org/wikipedia/donate/donationreminder/DonationReminderViewModel.kt

```diff
@@ -0,0 +1,195 @@
+        _uiState.update { it.copy(donationAmount = it.donationAmount.copy(selectedValue = donationAmount)) }
+    }
+
+    fun updateReadFrequencyState(readFrequency: Int) {
+        _uiState.update { it.copy(readFrequency = it.readFrequency.copy(selectedValue = readFrequency)) }
+    }
+
+    fun toggleDonationReminders(enabled: Boolean) {
+        Prefs.donationReminderConfig = Prefs.donationReminderConfig.copy(isEnabled = enabled)
+        if (enabled) {
+            Prefs.donationReminderConfig = Prefs.donationReminderConfig.copy(
+                initialPromptActive = false,
+                finalPromptActive = false
+            )
+        } else {
+            DonorExperienceEvent.logDonationReminderAction(
+                activeInterface = "global_setting",
+                action = "reminder_set_off"
+            )
+        }
+        _uiState.update { it.copy(isDonationReminderEnabled = enabled) }
+    }
+
+    private fun createReadFrequencyOptions(): SelectableOption<Int> {
+        val context = WikipediaApp.instance
+        val options = DonationReminderHelper.defaultReadFrequencyOptions
+        val optionItems = options.map {
+            OptionItem.Preset(it, context.resources.getQuantityString(R.plurals.donation_reminders_text_articles,
+                it, it))
+        } + OptionItem.Custom()
+
+        val selectedValue = if (Prefs.donationReminderConfig.articleFrequency <= 0) options.first()
+        else Prefs.donationReminderConfig.articleFrequency
+
+        return SelectableOption(
+            selectedValue,
+            optionItems,
+            minimumAmount = minArticleFrequencyLimit,
+            maximumAmount = maxArticleFrequencyLimit,
+            defaultValue = options.first(),
+            displayFormatter = {
+                context.resources.getQuantityString(R.plurals.donation_reminders_text_articles,
+                    it, it)
+            }
+        )
+    }
+
+    private suspend fun createDonationAmountOptions(): SelectableOption<Float> {
+        val donationConfig = DonationConfigHelper.getConfig()
+        val currencyCode = DonateUtil.currencyCode
+        val currentCountryCode = DonateUtil.currentCountryCode
+        val minimumAmount = donationConfig?.currencyMinimumDonation?.get(currencyCode) ?: 0f
+
+        var maximumAmount = donationConfig?.currencyMaximumDonation?.get(currencyCode) ?: 0f
+        if (maximumAmount == 0f) {
+            val defaultMin = donationConfig?.currencyMinimumDonation?.get(GooglePayComponent.CURRENCY_FALLBACK) ?: 0f
+            if (defaultMin > 0f) {
+                maximumAmount = (donationConfig?.currencyMinimumDonation?.get(currencyCode) ?: 0f) / defaultMin *
+                        (donationConfig?.currencyMaximumDonation?.get(GooglePayComponent.CURRENCY_FALLBACK) ?: 0f)
+            }
+        }
+
+        val presets = DonationReminderHelper.currencyAmountPresets[currentCountryCode] ?: listOf(0f)
```

Comment:
Instead of `0` as a fallback here, let's use `minimumAmount`, to reduce the chance of having a case where `0 €` is a selectable option.